### ISSUE TITLE
platform cluster radical

### ISSUE THREAD

**Issue by qazxiaye** [simgrid/simgrid#116]
Hello,

There are multiple clusters in my platform, when some clusters have identical radical (or same number of hosts in clusters), there will be an error msg like:

[0.000000] SimGrid-3.13/src/simix/smx_environment.cpp:42: [xbt/CRITICAL] Error while loading platform.xml: The route between Cloud-1_router and POP-0 already exists (Rq: routes are symmetrical by default).

It works when I modify one of the radicals.
Here is my platform : [https://drive.google.com/open?id=0B5mDEdbPR6y3R3BZRXRBMXdFSVU](https://drive.google.com/open?id=0B5mDEdbPR6y3R3BZRXRBMXdFSVU)

**Comment by adegomme** [comment]
Hello, and thanks for the report
Just to be sure, is the problem present with a platform not using the floyd routing model ?

**Comment by qazxiaye** [comment]
Hi,
The floyd routing model has been used in the platform.

**Comment by mquinson** [comment]
For the record, here is the platform file:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE platform SYSTEM "http://simgrid.gforge.inria.fr/simgrid/simgrid.dtd">
<platform version="4">
    <AS id="AS0" routing="Floyd">
        <host core="4" id="POP-0" speed="15Mf"/>
        <host core="4" id="POP-1" speed="18Mf"/>
        <cluster bw="1e9Bps" core="8" id="Cloud-0" lat="5E-5s" prefix="Cloud-0-" radical="1-4" router_id="Cloud-0_router" speed="969Mf" suffix=""/>
        <cluster bw="1e9Bps" core="4" id="Cloud-1" lat="5E-5s" prefix="Cloud-1-" radical="3-6" router_id="Cloud-1_router" speed="836Mf" suffix=""/>

        <link bandwidth="1000Mbps" id="l_5" latency="100us"/>
        <link bandwidth="1000Mbps" id="l_6" latency="100us"/>
        <link bandwidth="1000Mbps" id="l_7" latency="100us"/>
        <link bandwidth="1000Mbps" id="l_8" latency="100us"/>

        <route dst="POP-0" src="Cloud-0_router">
            <link_ctn id="l_5"/>
        </route>
        <route dst="POP-1" src="Cloud-0_router">
            <link_ctn id="l_6"/>
        </route>
        <route dst="POP-0" src="Cloud-1_router">
            <link_ctn id="l_7"/>
        </route>
        <route dst="POP-1" src="Cloud-1_router">
            <link_ctn id="l_8"/>
        </route>
    </AS>
</platform>
```

**Comment by frs69wq** [comment]
Changing the radicals gives the same parsing error to me.

I think the issue rather comes from the fact that POP-0 and the hosts withing Cloud-0 are in two different zones (the new name for ASes). A similar platform can be found here https://github.com/simgrid/simgrid/blob/master/examples/platforms/cluster_and_one_host.xml
The solution is to put POP-0 (and POP-1) in their own <zone> and then define routes between zones. A modified version of your platform would then be (in the 4.1 version of the DTD):

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE platform SYSTEM "http://simgrid.gforge.inria.fr/simgrid/simgrid.dtd">
<platform version="4.1">
  <zone id="zone0" routing="Floyd">
    <zone id="zone_POP_0" routing="None">
      <host core="4" id="POP-0" speed="15Mf"/>
    </zone>
    <zone id="zone_POP_1" routing="None">
      <host core="4" id="POP-1" speed="18Mf"/>
    </zone>
        <cluster bw="1e9Bps" core="8" id="Cloud-0" lat="5E-5s" prefix="Cloud-0-" radical="1-4" router_id="Cloud-0_router" speed="969Mf" suffix=""/>
        <cluster bw="1e9Bps" core="4" id="Cloud-1" lat="5E-5s" prefix="Cloud-1-" radical="3-6" router_id="Cloud-1_router" speed="836Mf" suffix=""/>

        <link bandwidth="1000Mbps" id="l_5" latency="100us"/>
        <link bandwidth="1000Mbps" id="l_6" latency="100us"/>
        <link bandwidth="1000Mbps" id="l_7" latency="100us"/>
        <link bandwidth="1000Mbps" id="l_8" latency="100us"/>

        <zoneRoute dst="zone_POP_0" src="Cloud-0" gw_src="Cloud-0_router" gw_dst="POP-0">
            <link_ctn id="l_5"/>
        </zoneRoute>
        <zoneRoute dst="zone_POP_1" src="Cloud-0" gw_src="Cloud-0_router" gw_dst="POP-1">
            <link_ctn id="l_6"/>
        </zoneRoute>
        <zoneRoute dst="zone_POP_0" src="Cloud-1" gw_src="Cloud-1_router" gw_dst="POP-0">
            <link_ctn id="l_7"/>
        </zoneRoute>
        <zoneRoute dst="zone_POP_1" src="Cloud-1" gw_src="Cloud-1_router" gw_dst="POP-1">
            <link_ctn id="l_8"/>
        </zoneRoute>
    </zone>
</platform>

Would this change be acceptable to you?

**Comment by mquinson** [comment]
This is actually a duplicate of #296: it makes no sense to mix hosts and cluster in the same zone, since the former can only appear in leafs of the routing tree while cluster are inner nodes. This got unnoticed for a while because I fixed #296 for <zone> while this example uses <AS>. Now, I fixed it for both forms of this tag, thus closing.

**Comment by mquinson** [comment]
Now, this platform file produces an error message as it should:
```
Invalid XML (XML input line 7, state 24): Starting tag <cluster> is not allowed here.
```
